### ISSUE TITLE
Force TypeScript only defineEmits

### ISSUE THREAD

**Issue by mesqueeb** [vuejs/eslint-plugin-vue#1960]
**Please describe what the rule should do:**
Forces the **TypeScript only** way of defining emits.

**What category should the rule belong to?**

[x] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

GOOD
```vue
const emit = defineEmits<{
  /**
   * The click event
   */
  (e: 'click', event: PointerEvent): void
}>()
```

BAD
```vue
const emit = defineEmits({
  /**
   * The click event
   */
  click: (event: PointerEvent) => !!event
})
```

BAD
```vue
const emit = defineEmits([
  /**
   * The click event
   */
  'click'
])
```

**Comment by FloEdelmann** [comment]
Similar to #1959.
